Library/Homebrew/os/linux/elf.rb

```diff
@@ -87,15 +87,16 @@ def with_interpreter?
   end
 
   def dynamic_elf?
-    return @dynamic_elf if defined? @dynamic_elf
-
-    @dynamic_elf = if which "readelf"
-      Utils.popen_read("readelf", "-l", to_path).include?(" DYNAMIC ")
-    elsif which "file"
-      !Utils.popen_read("file", "-L", "-b", to_path)[/dynamic|shared/].nil?
-    else
-      raise "Please install either readelf (from binutils) or file."
-    end
+    @dynamic_elf ||=
```

Comment:
This caching of the result does not work as expected for Boolean variables. A `false` result is effectively not cached.
@MikeMcQuaid What's the correct idiom to cache a Boolean variable?

Reply:
@sjackman  reset to old conditional ? `return @var unless @var.nil?`

Reply:
`return @var if defined?(@var)`

Reply:
i've restored to `if defined?(@var)`.

but can i have a clarification on why `defined?(@var) ` preferred over `@var.nil?`

Reply:
Not sure if it's @MikeMcQuaid's reason, but `defined?` more explicitly reflects what's happening here -- we're checking whether the variable has already been defined, not whether it's nil-ish.